src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/checkout/begin.js

```diff
@@ -7,43 +7,52 @@ const URLUtils = require('dw/web/URLUtils');
 const AdyenHelper = require('*/cartridge/scripts/util/adyenHelper');
 const { updateSavedCards } = require('*/cartridge/scripts/updateSavedCards');
 
-function begin(req, res, next) {
-  if (req.currentCustomer.raw.isAuthenticated()) {
-    updateSavedCards({
-      CurrentCustomer: req.currentCustomer.raw,
+function failCreatedOrder(cachedOrderNumber) {
+  const currentOrder = OrderMgr.getOrder(cachedOrderNumber);
+  // if order status is CREATED we can fail it and restore basket
+  if (currentOrder.status.value === Order.ORDER_STATUS_CREATED) {
+    Transaction.wrap(() => {
+      currentOrder.trackOrderChange(
+        'Failing order so cart can be restored; Shopper navigated back to checkout during payment redirection',
+      );
+      OrderMgr.failOrder(currentOrder, true);
     });
+    return true;
   }
-
+  return false;
+}
+function restoreBasket(cachedOrderNumber) {
   // restore cart if order number was cached
   try {
-    const cachedOrderNumber = req.session.privacyCache.get(
-      'currentOrderNumber',
-    );
+    let createdOrderFailed = false;
     if (cachedOrderNumber !== undefined) {
       const currentBasket = BasketMgr.getCurrentBasket();
-      const currentOrder = OrderMgr.getOrder(cachedOrderNumber);
-      // if current basket is null or empty
-      if (
-        !currentBasket ||
-        currentBasket.getAllProductLineItems().length === 0
-      ) {
-        // if order status is CREATED we can fail it and restore basket
-        if (currentOrder.status.value === Order.ORDER_STATUS_CREATED) {
-          Transaction.wrap(() => {
-            currentOrder.trackOrderChange(
-              'Failing order so cart can be restored; Shopper navigated back to checkout during payment redirection',
-            );
-            OrderMgr.failOrder(currentOrder, true);
-          });
-          const emit = (route) => this.emit(route, req, res);
-          res.redirect(URLUtils.url('Checkout-Begin', 'stage', 'shipping'));
-          emit('route:Complete');
-          return true;
-        }
+      if (currentBasket?.getAllProductLineItems().length > 0) {
+        return createdOrderFailed;
```

Comment:
I suggest we return false here, easier to read

Reply:
This should be `true` :D